### ISSUE TITLE
page_builder#add method does not allow an array that includes nil

### ISSUE THREAD

**Issue by muziyoshiz** [embulk/embulk#219]
In Embulk v0.6.13 and v0.6.14, if page_builder#add method received an array that included nil in parser plugins, following exception raised. Such an array was allowed in previous versions of Embulk.

If nil was replaced by "" before calling page_builder#add method, the following exception did not raise.

```
2015-06-30 00:39:27.546 +0900 [INFO] (preview): Loading files [/Users/myoshiz/devel/myproject/myproject.log.20150314]
java.lang.NullPointerException
        at org.embulk.spi.util.dynamic.AbstractDynamicColumnSetter.setRubyObject(org/embulk/spi/util/dynamic/org/embulk/spi/util/dynamic/org/embulk/spi/util/dynamic/AbstractDynamicColumnSetter.java:76)
        at java.lang.reflect.Method.invoke(java/lang/reflect/java/lang/reflect/java/lang/reflect/Method.java:483)
        at RUBY.add(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/page_builder.rb:19)
        at RUBY.add(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/page_builder.rb:19)
        at RUBY.add(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/page_builder.rb:19)
        at RUBY.add(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/page_builder.rb:19)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:109)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:109)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:109)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:109)
        at org.jruby.RubyKernel.loop(org/jruby/org/jruby/org/jruby/RubyKernel.java:1507)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:91)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:91)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:91)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:91)
        at org.jruby.RubyKernel.loop(org/jruby/org/jruby/org/jruby/RubyKernel.java:1507)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:52)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:52)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:52)
        at RUBY.run(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:52)
        at RUBY.run(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:57)
        at RUBY.run(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:57)
        at RUBY.run(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:57)
        at RUBY.run(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:57)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.run(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.run(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.run(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.run(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.run(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.run(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.run(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.run(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at org.embulk.spi.FileInputRunner.run(org/embulk/spi/org/embulk/spi/org/embulk/spi/FileInputRunner.java:145)
        at org.embulk.exec.PreviewExecutor$2$1.run(org/embulk/exec/org/embulk/exec/org/embulk/exec/PreviewExecutor.java:107)
        at org.embulk.spi.util.Filters$RecursiveControl.transaction(org/embulk/spi/util/org/embulk/spi/util/org/embulk/spi/util/Filters.java:97)
        at org.embulk.spi.util.Filters.transaction(org/embulk/spi/util/org/embulk/spi/util/org/embulk/spi/util/Filters.java:50)
        at org.embulk.exec.PreviewExecutor$2.run(org/embulk/exec/org/embulk/exec/org/embulk/exec/PreviewExecutor.java:96)
        at org.embulk.spi.FileInputRunner$RunnerControl$1$1.run(org/embulk/spi/org/embulk/spi/org/embulk/spi/FileInputRunner.java:117)
        at java.lang.reflect.Method.invoke(java/lang/reflect/java/lang/reflect/java/lang/reflect/Method.java:483)
        at RUBY.transaction(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:46)
        at RUBY.transaction(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:46)
        at RUBY.transaction(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:46)
        at RUBY.transaction(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:46)
        at RUBY.transaction(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:38)
        at RUBY.transaction(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:38)
        at RUBY.transaction(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:38)
        at RUBY.transaction(/Users/myoshiz/devel/embulk-parser-multiline-log-sample/lib/embulk/parser/multiline-log-sample.rb:38)
        at RUBY.transaction(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:43)
        at RUBY.transaction(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:43)
        at RUBY.transaction(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:43)
        at RUBY.transaction(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/parser_plugin.rb:43)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.transaction(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.transaction(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.transaction(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.transaction(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.transaction(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.transaction(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.transaction(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at Embulk$$ParserPlugin$$JavaAdapter_1742249157.transaction(Embulk$$ParserPlugin$$JavaAdapter_1742249157.gen:13)
        at org.embulk.spi.FileInputRunner$RunnerControl$1.run(org/embulk/spi/org/embulk/spi/org/embulk/spi/FileInputRunner.java:111)
        at org.embulk.spi.util.Decoders$RecursiveControl.transaction(org/embulk/spi/util/org/embulk/spi/util/org/embulk/spi/util/Decoders.java:77)
        at org.embulk.spi.util.Decoders.transaction(org/embulk/spi/util/org/embulk/spi/util/org/embulk/spi/util/Decoders.java:33)
        at org.embulk.spi.FileInputRunner$RunnerControl.run(org/embulk/spi/org/embulk/spi/org/embulk/spi/FileInputRunner.java:108)
        at org.embulk.standards.LocalFileInputPlugin.resume(org/embulk/standards/org/embulk/standards/org/embulk/standards/LocalFileInputPlugin.java:80)
        at org.embulk.standards.LocalFileInputPlugin.transaction(org/embulk/standards/org/embulk/standards/org/embulk/standards/LocalFileInputPlugin.java:70)
        at org.embulk.spi.FileInputRunner.transaction(org/embulk/spi/org/embulk/spi/org/embulk/spi/FileInputRunner.java:63)
        at org.embulk.exec.PreviewExecutor.doPreview(org/embulk/exec/org/embulk/exec/org/embulk/exec/PreviewExecutor.java:93)
        at org.embulk.exec.PreviewExecutor.access$000(org/embulk/exec/org/embulk/exec/org/embulk/exec/PreviewExecutor.java:27)
        at org.embulk.exec.PreviewExecutor$1.run(org/embulk/exec/org/embulk/exec/org/embulk/exec/PreviewExecutor.java:67)
        at org.embulk.exec.PreviewExecutor$1.run(org/embulk/exec/org/embulk/exec/org/embulk/exec/PreviewExecutor.java:63)
        at org.embulk.spi.Exec.doWith(org/embulk/spi/org/embulk/spi/org/embulk/spi/Exec.java:24)
        at org.embulk.exec.PreviewExecutor.preview(org/embulk/exec/org/embulk/exec/org/embulk/exec/PreviewExecutor.java:63)
        at org.embulk.command.Runner.preview(org/embulk/command/org/embulk/command/org/embulk/command/Runner.java:250)
        at org.embulk.command.Runner.main(org/embulk/command/org/embulk/command/org/embulk/command/Runner.java:110)
        at java.lang.reflect.Method.invoke(java/lang/reflect/java/lang/reflect/java/lang/reflect/Method.java:483)
        at RUBY.run(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/command/embulk_run.rb:351)
        at RUBY.run(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/command/embulk_run.rb:351)
        at RUBY.run(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/command/embulk_run.rb:351)
        at RUBY.run(file:/Users/myoshiz/.embulk/bin/embulk!/embulk/command/embulk_run.rb:351)
        at classpath_3a_embulk.command.embulk.(root)(classpath:embulk/command/embulk.rb:47)
        at classpath_3a_embulk.command.embulk.(root)(classpath_3a_embulk/command/classpath:embulk/command/embulk.rb:47)
        at classpath_3a_embulk.command.embulk.(root)(classpath_3a_embulk/command/classpath:embulk/command/embulk.rb:47)
        at classpath_3a_embulk.command.embulk.(root)(classpath_3a_embulk/command/classpath_3a_embulk/command/classpath:embulk/command/embulk.rb:47)
        at classpath_3a_embulk.command.embulk.(root)(classpath_3a_embulk/command/classpath:embulk/command/embulk.rb:47)
        at classpath_3a_embulk.command.embulk.(root)(classpath_3a_embulk/command/classpath_3a_embulk/command/classpath:embulk/command/embulk.rb:47)
        at classpath_3a_embulk.command.embulk.(root)(classpath_3a_embulk/command/classpath_3a_embulk/command/classpath:embulk/command/embulk.rb:47)
        at classpath_3a_embulk.command.embulk.(root)(classpath_3a_embulk/command/classpath_3a_embulk/command/classpath_3a_embulk/command/classpath:embulk/command/embulk.rb:47)
        at org.embulk.cli.Main.main(org/embulk/cli/org/embulk/cli/org/embulk/cli/Main.java:11)
```

Sample plugin used in the above case: https://github.com/muziyoshiz/embulk-parser-multiline-log-sample/blob/master/lib/embulk/parser/multiline-log-sample.rb

**Comment by frsyuki** [comment]
Fixed at #222!

**Comment by muziyoshiz** [comment]
Embulk v0.6.15 solved the problem in my enviroment. Thank you!
